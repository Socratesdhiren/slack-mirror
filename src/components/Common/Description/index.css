.term {
  display: table-cell;
  color: rgba(0, 0, 0, 0.85);
}

.term:after {
  content: ':';
  margin: 0 8px 0 2px;
  position: relative;
  top: -0.5px;
}

.detail {
  margin-left: 10px;
  padding-bottom: 16px;
}
