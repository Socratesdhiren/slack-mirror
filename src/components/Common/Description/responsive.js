export default {
  1: { xs: 24 },
  2: { xs: 24, sm: 12 },
  3: { xs: 24, sm: 12, md: 8 },
  4: { xs: 24, sm: 12, md: 8 },
  5: { lg: 8, xl: 6, md: 8 },
};
