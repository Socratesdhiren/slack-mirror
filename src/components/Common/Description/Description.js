import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'antd';
import './index.css';
import responsive from './responsive';

const Description = ({ term, column, children }) => (
  <Row type="flex">
    <Col {...responsive[column]}>{term && <div className="term">{term}</div>}</Col>
    <Col xl={{ span: 12 }} lg={{ span: 12 }} md={{ span: 12 }} sm={16}>
      {children !== null && children !== undefined && <div className="detail">{children}</div>}
    </Col>
  </Row>
);

Description.defaultProps = {
  term: '',
};

Description.propTypes = {
  term: PropTypes.node,
};

export default Description;
