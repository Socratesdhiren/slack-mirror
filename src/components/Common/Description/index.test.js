import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Description from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Component --- Description', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Description />);
  });

  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
