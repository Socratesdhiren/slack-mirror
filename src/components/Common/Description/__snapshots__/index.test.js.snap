// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component --- Description should render correctly 1`] = `
<Row
  gutter={0}
  type="flex"
>
  <Col />
  <Col
    lg={
      Object {
        "span": 12,
      }
    }
    md={
      Object {
        "span": 12,
      }
    }
    sm={16}
    xl={
      Object {
        "span": 12,
      }
    }
  />
</Row>
`;
